### PR TITLE
Change "Golioth SDK" to "Golioth Zephyr SDK"

### PR DESCRIPTION
**Pull request by 0Grit**
@mniestroj Any other references we should update to indicate this is specifically the Zephyr SDK?

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -1,15 +1,15 @@
-Getting Started
+Golioth Zephyr SDK
 ***************
 
-Using with Zephyr RTOS
+Getting Started
 ======================
```
Comment (mniestroj) [inline README.rst:5]: Line endorsement length should be the same as the section title length. Same in other places.

File: README.rst [hunk 2]
```diff
@@ -62,7 +62,7 @@ Add following entry to ``west.yml`` file in ``manifest/projects`` subtree:
     - name: golioth
       path: modules/lib/golioth
       revision: main
-      url: git@github.com:golioth/zephyr.git
+      url: git@github.com:golioth/zephyr-sdk.git
```
Comment (mniestroj) [inline README.rst:65]: There is another link to update in section "Adding Golioth SDK for Zephyr to existing west project".

Comment (rajeev1986) [reply README.rst:65]: Thanks for catching this. I came across this discrepancy while installing Golioth SDK this weekend and also mentioned it to @beriberikix. :)

**Comment by mniestroj** [comment]
@0Grit there are some merge conflicts and also unresolved comments.

**Comment by mniestroj** [comment]
@ChrisGammell @szczys Do we need those changes? If yes, could you update this PR or create a new one? If not, then let's just close this PR.

**Pull request closed without merging**